java-connectors/kafka-connect-azure-servicebus/src/main/java/io/lenses/streamreactor/connect/azure/servicebus/source/TaskToReceiverBridge.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2017-2024 Lenses.io Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.lenses.streamreactor.connect.azure.servicebus.source;
+
+import io.lenses.kcql.Kcql;
+import io.lenses.streamreactor.connect.azure.servicebus.config.AzureServiceBusConfigConstants;
+import io.lenses.streamreactor.connect.azure.servicebus.util.KcqlConfigBusMapper;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.apache.kafka.connect.storage.OffsetStorageReader;
+
+/**
+ * Bridge between Receivers and Connector's Task for Azure Service Bus.
+ */
+@Slf4j
+public class TaskToReceiverBridge {
+
+  private static final int INITIAL_RECORDS_TO_COMMIT_SIZE = 500;
+  private final BlockingQueue<ServiceBusMessageHolder> recordsQueue;
+  private final ServiceBusPartitionOffsetProvider offsetProvider;
```

Comment:
offsetProvider is written but never accessed

Reply:
Then the branching will still have to happen, but will be in `SourceTask` which essentially doesn't have to know about Bridge's implementation details (that it initializes Facades).

I'd advise leaving it as-is.

Reply:
Removed